packages/legend-application-studio/src/stores/editor-state/GrammarTextEditorState.ts

```diff
@@ -183,7 +184,14 @@ export class GrammarTextEditorState {
     }
     this.currentElementLabelRegexString = getGrammarElementTypeLabelRegexString(
       typeLabel,
-      element.path,
+      `${element.package?.path}::${generatePackageableElementTreeNodeDataLabel(
+        element,
+      )
+        .replaceAll('*', '\\*')
```

Comment:
what are we trying to replace here?

Reply:
make sure we could identify *, [, ], (, )  when using regex

Reply:
shouldnt the logic then live in `getGrammarElementTypeLabelRegexString`